internal/controller/monovertexrollout/monovertexrollout_progressive.go

```diff
@@ -315,6 +315,9 @@ func scaleDownUpgradingMonoVertex(
 		// which is necessary when performing the health check for "ready replicas >= desired replicas"
 		// 2. that the total number of Pods (between the 2 monovertices) before and during upgrade remains the same
 		upgradingChildScaleTo := scaleValue.Initial - scaleValue.ScaleTo
+		if upgradingChildScaleTo <= 0 { // if for some reason the Initial value was 0, we don't want to set our Pods to 0
+			upgradingChildScaleTo = 1
+		}
```

Comment:
this was a fix made to Pipeline that I'm now incorporating into Monovertex as well.